### DISCUSSION TITLE
Documentation with the default key bindings

### DISCUSSION THREAD

**Discussion by balta2ar** [marlonrichert/zsh-autocomplete#205] [Q&A] [answered]
It would be nice if README contained a set of default keybindings (https://github.com/marlonrichert/zsh-autocomplete#advanced-choose-your-own-key-bindings) that are on by default, but which you can copy and partially disable (good example: https://github.com/neoclide/coc.nvim#example-vim-configuration).

Right now the docs says `You can then use zsh-autocomplete's widgets to define your own key bindings` probably assuming that I know how to do that like the back of my hand. Right now, for example, I'd like to disable only one keybinding (Tab). Copying sample configuration and commenting out one line would've been faster for me.

Thanks!

**Comment by marlonrichert** [comment]
> Right now, for example, I'd like to disable only one keybinding (Tab).

Why do you want to disable Tab? What exactly would you like Tab to do?

**Comment by balta2ar** [comment]
Well, I like some of the experience that zsh-autocomplete gives me, but not all of it. I ended up disabling Tab, Up and Down functionality for now and living with the rest for a few days.

Speaking specifically about Tab: it conflicts with fzf (see how `**` works in https://github.com/junegunn/fzf/wiki/Examples-(completion) and https://github.com/junegunn/fzf/wiki/Configuring-fuzzy-completion) and I use it sometimes, thus I'd prefer my default Tab behavior. And this is totally fine -- you can't have cake and eat it too :)

Same with Up/Down -- it adds 1 extra keypress to my regular workflow. To enter the menu that zsh-autocomplete offers I currently use PageDown and it works for me.

**Comment by marlonrichert** [comment, marked as answer]
You can use Fzf's Tab widget instead of `zsh-autocomplete`'s by adding the following to your `~/.zshrc` file:
```zsh
zstyle ':autocomplete:tab:*' fzf yes
```
This is documented here: https://github.com/marlonrichert/zsh-autocomplete/blob/master/README.md#change-tab-behavior

For all other key bindings, with commit edf6f867fa75032ae55403891dde014bb33ca1f4, I've made it possible to override them manually. You can now find instructions with examples for this over here: https://github.com/marlonrichert/zsh-autocomplete/blob/master/README.md#change-other-key-bindings

For your case specifically, this is probably what you want:
```zsh
source path/to/zsh-autocomplete
# zsh-autocomplete defines an associative array `key` that you can use.
# The following lines revert the given keys back to Zsh defaults.
bindkey $key[Up] up-line-or-history
bindkey $key[Down] down-line-or-history
```

**Comment by balta2ar** [comment]
This is awesome, thank you so much for your work!

**Comment by balta2ar** [comment]
@marlonrichert There is one minor nitpick that I just noticed after a day of using it. When zsh-autocomplete is off, when I press Tab and when there are multiple ambiguous candidates, the command is completed up to the point of unambiguity, and then it stops. The second Tab press reveals a menu. The third Tab  activates the menu where you can navigate and pick a candidate.

However, when I enable zsh-autocomplete and when I use settings that you suggested (`zstyle ':autocomplete:tab:*' fzf yes`), when I press Tab for the first time, (and assuming there is no fzf `**` trigger there) the CLI is fully expanded with the first candidate. Which sort of saves me two extra Tab presses, but the reality is that the first candidate is often not what I want, so I have to erase part of the string and start again.

I wonder if it's possible to have both fzf as a priority (have opiton `zstyle ':autocomplete:tab:*' fzf yes`) but also have the behavior from the first paragraph (with Three tabs).

**Comment by marlonrichert** [comment]
Yes, it's all in the Readme, in the same section: https://github.com/marlonrichert/zsh-autocomplete/blob/master/README.md#change-tab-behavior

What you want is this:
```zsh
zstyle ':autocomplete:tab:*' insert-unambiguous yes
zstyle ':autocomplete:tab:*' widget-style menu-select
zstyle ':autocomplete:tab:*' fzf yes
```

Although, I have to add: You can usually get the completion you want with the default settings by just continuing to type ahead. For example, given the following state:
```zsh
> git ch
check-attr        display gitattributes information
check-ignore      debug gitignore/exclude files
check-mailmap     show canonical names and email addresses of contacts
check-ref-format  ensure that a reference name is well formed
checkout          checkout branch or paths to working tree
checkout-index    copy files from index to working directory
cherry            find commits not merged upstream
cherry-pick       apply changes introduced by some existing commits
```

If you type just two more letters, you can already get what you want. For example:
```zsh
> git chou
checkout        checkout branch or paths to working tree
checkout-index  copy files from index to working directory
common substring:  ⇤ Shift-Tab to insert
checkout
```

Or:
```zsh
> git chpi
cherry-pick  apply changes introduced by some existing commits
```

I find that's using the default settings and typing a couple more letters is usually faster than doing the whole Tab dance you just described.
